### ISSUE TITLE
anonymous function name behave differently from V8

### ISSUE THREAD

**Issue by alexlamsl** [nodejs/node-chakracore#201]
From https://github.com/mishoo/UglifyJS2/issues/1703
Tested `chakracore-nightly 8.0.0-nightly20170325a35216f4b2`

Test code:
```js
var b = 10;

!function a() {
    a++;
    b = a;
}();

console.log(b);
```

Console output:
```sh
D:\UglifyJS2> nvs use node
PATH = node\6.10.1\x64

D:\UglifyJS2> type test.js | node
[Function: a]

D:\UglifyJS2> nvs use chakracore-nightly
PATH = chakracore-nightly\8.0.0-nightly20170325a35216f4b2\x64

D:\UglifyJS2> type test.js | node
NaN
```

**Comment by alexlamsl** [comment]
With this minimal addition to the example above, `node-chakracore` will match the behaviour of `node-v8`:
```js
var b = 10;

!function a() {
    a++;
    b = a;
    !function() { a; }();
}();

console.log(b);
```

Console output:
```sh
D:\UglifyJS2> nvs use node
PATH = node\6.10.1\x64

D:\UglifyJS2> type test.js | node
[Function: a]

D:\UglifyJS2> nvs use chakracore-nightly
PATH = chakracore-nightly\8.0.0-nightly20170325a35216f4b2\x64

D:\UglifyJS2> type test.js | node
[Function: a]
```

**Comment by alexlamsl** [comment]
Reproducible as of `8.4.0`
```sh
D:\> nvs use node
PATH = node\8.4.0\x64

D:\> type test.js | node
[Function: a]

D:\> nvs use chakracore
PATH = chakracore\8.4.0\x64

D:\> type test.js | node
NaN
```

**Comment by alexlamsl** [comment]
Still an issue as of `v10.1.0`

**Comment by alexlamsl** [comment]
Fixed in `v10.6.0` - thanks!
